dolphin/db/api.py

```diff
@@ -59,56 +58,207 @@
+                      'desc' for descending order
+    :param filters: dictionary of filters
+    :param offset: number of items to skip
+    :returns: list of storage pools
+    """
+    return IMPL.pool_get_all(context, marker, limit, sort_keys, sort_dirs,
+                             filters, offset)
+
+
+def disk_create(context, values):
+    """Create a disk from the values dictionary."""
+    return IMPL.disk_create(context, values)
 
-def registry_context_create(values):
-    return IMPL.registry_context_create(values)
 
+def disk_update(context, disk_id, values):
+    """Update a disk withe the values dictionary."""
+    return IMPL.disk_create(context, disk_id, values)
 
-def registry_context_update(values):
-    return IMPL.registry_context_update(values)
 
+def disk_get(context, disk_id):
+    """Get a disk or raise an exception if it does not exist."""
+    return IMPL.disk_get(context, disk_id)
 
-def registry_context_get(storage_id):
-    return IMPL.registry_context_get(storage_id)
 
+def disk_get_all(context, marker=None, limit=None, sort_keys=None,
+                 sort_dirs=None, filters=None, offset=None):
+    """Retrieves all disks.
 
-def registry_context_get_all():
-    return IMPL.registry_context_get_all()
+    If no sort parameters are specified then the returned volumes are sorted
+    first by the 'created_at' key and then by the 'id' key in descending
+    order.
+
+    :param context: context of this request, it's helpful to trace the request
+    :param marker: the last item of the previous page, used to determine the
+                   next page of results to return
+    :param limit: maximum number of items to return
+    :param sort_keys: list of attributes by which results should be sorted,
+                      paired with corresponding item in sort_dirs
+    :param sort_dirs: list of directions in which results should be sorted,
+                      paired with corresponding item in sort_keys, for example
+                      'desc' for descending order
+    :param filters: dictionary of filters
+    :param offset: number of items to skip
+    :returns: list of disks
+    """
+    return IMPL.disk_get_all(context, marker, limit, sort_keys, sort_dirs,
+                             filters, offset)
+
+
+def registry_context_create(context, values):
+    """Create a registry context that used to connect
+    to a specific storage device.
+    """
+    return IMPL.registry_context_create(context, values)
+
+
+def registry_context_update(context, registry_context_id, values):
+    """Update a registry context with the values dictionary."""
+    return IMPL.registry_context_update(context, registry_context_id, values)
+
+
+def registry_context_get(context, storage_id):
+    """Get a registry context."""
+    return IMPL.registry_context_get(context, storage_id)
+
+
+def registry_context_get_all(context, marker=None, limit=None, sort_keys=None,
```

Comment:
can you change 'sort_keys' to  'sort_key' and 'sort_dirs' to 'sort_dir'.
beacuse it will be a single value.

Reply:
Eh... I think some times query order by multiple items is needed, and database can support by more than one item, for example: SELECT * FROM table1 ORDER BY field1 ASC, field2 DESC. How do you think?

Reply:
ya, but in our API we support one sort_key , and one sort direction .. hope that is the standard..

Reply:
I think it's OK that our API only supports one sort_key,  this definition does not conflict with it. We can extend to support multiple sort_keys without changing database API in the future.